apps/astarte_housekeeping/lib/astarte_housekeeping/config.ex

```diff
@@ -94,6 +109,37 @@ defmodule Astarte.Housekeeping.Config do
     end
   end
 
+  @doc """
+  Returns :ok if at least one of HOUSEKEEPING_ASTARTE_KEYSPACE_REPLICATION_FACTOR or HOUSEKEEPING_ASTARTE_KEYSPACE_NETWORK_REPLICATION_MAP is valid, based on HOUSEKEEPING_ASTARTE_KEYSPACE_REPLICATION_STRATEGY value.
+  """
+  def validate_astarte_replication! do
+    case astarte_keyspace_replication_strategy!() do
+      :simple -> validate_astarte_replication_factor!()
+      :network -> validate_astarte_replication_map!()
+      nil -> raise "Invalid replication strategy set for the astarte keyspace"
+    end
+  end
+
+  defp validate_astarte_replication_factor! do
+    case astarte_keyspace_replication_factor() do
+      {:ok, _} ->
+        :ok
+
+      _ ->
+        raise "Invalid or empty replication factor for the astarte keyspace. Check the value of HOUSEKEEPING_ASTARTE_KEYSPACE_REPLICATION_FACTOR"
```

Comment:
technically the "empty" case is not a problem here as the default would be used. it is indeed a problem for the replication map though

Let's also make it clear that we're expecting simple replication to be used here

```suggestion
        raise "Invalid replication factor for the astarte keyspace with simple replication strategy. Check the values of HOUSEKEEPING_ASTARTE_KEYSPACE_REPLICATION_STRATEGY and HOUSEKEEPING_ASTARTE_KEYSPACE_REPLICATION_FACTOR"
```